pkg/execution/state/redis_state/lua/includes/update_account_queues.lua

```diff
@@ -0,0 +1,29 @@
+local function account_is_set(keyAccountPartitions)
+  return exists_without_ending(keyAccountPartitions, "accounts:00000000-0000-0000-0000-000000000000:partition:sorted") == true
+end
+
+-- get_fn_partition_score returns a fn's earliest job as a score for pointer queues.
+-- This returns 0 if there are no scores available.
+local function get_earliest_account_partition_score(keyAccountPartitions)
+    local earliestScore = redis.call("ZRANGE", keyAccountPartitions, "-inf", "+inf", "BYSCORE", "LIMIT", 0, 1, "WITHSCORES")
+    if earliestScore == nil or earliestScore == false or earliestScore[2] == nil then
+        return 0
+    end
+    -- pointers are already second precision, so we do not want to truncate any further (as opposed to get_fn_partition_score)
+    -- earliest is a table containing {item, score}
+    return tonumber(earliestScore[2])
+end
+
+-- This function updates account queues
+-- Requires: update_pointer_score.lua, ends_with.lua
+local function update_account_queues(keyGlobalAccountPointer, keyAccountPartitions, partitionID, accountId, score)
+  -- we might be leasing an "old" partition which doesn't store the account
+  if account_is_set(keyAccountPartitions) == true then
```

Comment:
TODO: Go over this once more to make sure this is sound

PartitionLease/PartitionRequeue act on old or new partitions. Old partitions are workflow partitions, new partitions are key queues. Old partitions might not store the account ID, so the key and value will be invalid.

As soon as we enqueue a new item, the right queues should be created (as we generate new partitions by using `ItemPartitions`), and the next PartitionLease will work on that.

So this case should only happen in new executors (running the new code) that process old partitions (that haven't enqueued a new item).

Reply:
Correct, and that should be a safe operation as account queues are all forward-compatible;  if you're not enqueueing, you don't get them.